### PR TITLE
New user experience

### PR DESCRIPTION
**Pull request by seanyboy49**
- [x] Adds overlays for new user experience
- [x] Adds `useLocalStorage` hook for convenience in using local storage
- [x] Adds `useNux` hook for nux callbacks

### CODE REVIEW

File: client/src/components/LightBox/index.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    img: <SwipeLeftImg />,
+  },
+  nuxSwipeRight: {
+    texts: ['SWIPE RIGHT', 'TO CHOOSE', 'THIS FILM'],
+    img: <SwipeRightImg />,
+  },
+  nuxSwipeDown: 'nuxSwipeDown',
+}
+
+export const categories = Object.keys(categoryMap).reduce((acc, curr) => {
```
Comment (seanyboy49) [inline client/src/components/LightBox/index.js:20]: Returns an object generated from categoryMap keys where
 the keys and values are the same
 example return value
```
 {
  nuxSwipeLeft: "nuxSwipeLeft"
  nuxSwipeRight: "nuxSwipeRight"
 }
```

File: client/src/components/LightBox/index.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+const categoryMap = {
+  nuxSwipeLeft: {
+    texts: ['SWIPE LEFT', 'TO PASS ON', 'THIS FILM'],
+    img: <SwipeLeftImg />,
+  },
+  nuxSwipeRight: {
+    texts: ['SWIPE RIGHT', 'TO CHOOSE', 'THIS FILM'],
+    img: <SwipeRightImg />,
+  },
+  nuxSwipeDown: 'nuxSwipeDown',
```
Comment (seanyboy49) [inline client/src/components/LightBox/index.js:17]: I will eventually add info for the swipe down experience

File: client/src/components/PosterStack/index.js [hunk 3]
```diff
@@ -1,29 +1,20 @@
...
 import { useSprings } from 'react-spring'
 import { useDrag } from 'react-use-gesture'
+import PropTypes from 'prop-types'
 
 import Poster from './Poster'
+import { to, from, trans } from './utility'
 import { StackContainer } from './styled'
+import { categories as lightBoxCategories } from '../LightBox'
 
-const to = (i) => ({
```
Comment (seanyboy49) [inline client/src/components/PosterStack/index.js:8]: moved these to a utility file to keep the file a little cleaner

File: client/src/components/PosterStack/index.js [hunk 4]
```diff
@@ -38,6 +29,32 @@ const PosterStack = ({ movies }) => {
       const dir = xDir < 0 ? -1 : 1
       if (!down && trigger) gone.add(index)
 
+      // Custom click handling for NUX interactions
+      if (onClick && Math.abs(mx) >= 20) {
```
Comment (seanyboy49) [inline client/src/components/PosterStack/index.js]: Lots of nested ifs...
I'll try to think of a better way of doing this

**Review by JinaZhu — APPROVED** [review]

**Pull request merged**